packages/playground/cli/src/run-cli.ts

```diff
@@ -530,6 +517,81 @@ export async function runCLI(args: RunCLIArgs): Promise<RunCLIServer> {
 				Number.MAX_SAFE_INTEGER / totalWorkerCount
 			);
 
+			/*
+			 * Use a real temp dir as a target for the following Playground paths
+			 * so that multiple worker threads can share the same files.
+			 *  - /internal
+			 *  - /tmp
+			 *  - /wordpress
+			 *
+			 * Sharing the same files leads to faster boot times and uses less memory
+			 * because we don't have to create or maintain multiple copies of the same files.
+			 */
+			const tempDirNameDelimiter = '-playground-cli-site-';
+			const nativeDirPath = await createPlaygroundCliTempDir(
+				tempDirNameDelimiter
+			);
+
+			// We do not know the system temp dir,
+			// but we can try to infer from the location of the current temp dir.
+			const tempDirRoot = path.dirname(nativeDirPath);
+
+			const twoDaysInMillis = 2 * 24 * 60 * 60 * 1000;
+			const tempDirStaleAgeInMillis = twoDaysInMillis;
+
+			// NOTE: This is an async operation, but we do not care to block on it.
+			// Let's let the cleanup happen as the main thread has time.
+			cleanupStalePlaygroundTempDirs(
+				tempDirNameDelimiter,
+				tempDirStaleAgeInMillis,
+				tempDirRoot
+			);
+
+			// NOTE: We do not add mount declarations for /internal here
+			// because it will be mounted as part of php-wasm init.
+			const nativeInternalDirPath = path.join(nativeDirPath, 'internal');
+			mkdirSync(nativeInternalDirPath);
+
+			const userProvidableNativeSubdirs = [
+				'wordpress',
+				// Note: These dirs are from Emscripten's "default dirs" list:
+				// https://github.com/emscripten-core/emscripten/blob/f431ec220e472e1f8d3db6b52fe23fb377facf30/src/lib/libfs.js#L1400-L1402
+				//
+				// Any Playground process with multiple workers may assume
+				// these are part of a shared filesystem, so let's recognize
+				// them explicitly here.
+				'tmp',
+				'home',
+			];
+
+			for (const subdirName of userProvidableNativeSubdirs) {
+				const isMountingSubdirName = (mount: Mount) =>
+					mount.vfsPath === `/${subdirName}`;
+				const thisSubdirDoesNotHaveAMount = !(
+					args['mount-before-install']?.some(isMountingSubdirName) ||
+					args['mount']?.some(isMountingSubdirName)
+				);
+				if (thisSubdirDoesNotHaveAMount) {
+					// The user isn't already mounting a native dir for this,
+					// so let's create a mount from within our native temp dir.
+					const nativeSubdirPath = path.join(
+						nativeDirPath,
+						subdirName
+					);
+					mkdirSync(nativeSubdirPath);
+
+					if (args['mount-before-install'] === undefined) {
+						args['mount-before-install'] = [];
+					}
+
+					// Make the real mount first so any further subdirs are mounted into it.
```

Comment:
Have you confirmed the nested mounts actually work?

Reply:
I'm not convinced about the implication of `auto-mount tests passing` => `nested mounts are supported`. Let's confirm that explicitly.